rasterio/_warp.pyx

```diff
@@ -338,13 +338,13 @@ def _reproject(
         # We need a src_transform and src_dst in this case. These will
         # be copied to the MEM dataset.
         if dtypes.is_ndarray(source):
-
             if not src_crs:
                 raise CRSError("Missing src_crs.")
-
             if src_nodata is None and hasattr(source, 'fill_value'):
                 # source is a masked array
```

Comment:
Is this okay?
```python
>>> import numpy
>>> ma = numpy.ma.array([1, 2], mask=[True, False])
>>> ma
masked_array(data=[--, 2],
             mask=[ True, False],
       fill_value=999999)
>>> numpy.array(ma, copy=False)
array([1, 2])
```

Reply:
@snowman2 I don't understand. Are you asking whether we will miss the mask?

Reply:
This is just something that stood out to me. I am pretty sure it is fine, but wanted to add a call out to for you to see it just to be sure it wouldn't break anything.

Reply:
It looks like this PR doesn't do anything about the masked array case. Does xarray not have a masked array?

Reply:
> Does xarray not have a masked array?

No, it doesn't. It converts integer to floats and uses NaN for the mask.

Reply:
https://corteva.github.io/rioxarray/stable/getting_started/nodata_management.html